### ISSUE TITLE
[Windows] Failed to package Android APK using Java 17

### ISSUE THREAD

**Issue by itlancer** [airsdk/Adobe-Runtime-Support#1368]
## Problem Description
Failed to package Android APK using Java 17 for Windows.

It has been tested with latest AIR SDK 33.1.1.674 with different Windows 10 64-bit devices and Java 17.
Same problem in all cases.
There is no such problems with Java 8 and Java 11.
Didn't test it for macOS environment.

## Steps to Reproduce
Install Java 17 for your Windows device and try to package AIR application for Android.
Packaging command:
`adt -package -target apk-captive-runtime -storetype pkcs12 -keystore android.p12 -storepass password -keypass password test_app.apk test_app-app.xml -extdir libs test_app.swf`

**Actual Result:**
Packaging error:
```
Exception in thread "main" java.lang.IllegalAccessError: class com.adobe.air.apk.APKSigner (in unnamed module @0x3bd40a57) cannot access class sun.security.x509.X500Name (in module java.base) because module java.base does not export sun.security.x509 to unnamed module @0x3bd40a57
	at com.adobe.air.apk.APKSigner.getSignatureBlockFile(APKSigner.java)
	at com.adobe.air.apk.APKOutputStream.finalizeSig(APKOutputStream.java)
	at com.adobe.air.ApplicationPackager.createPackage(ApplicationPackager.java)
	at com.adobe.air.ADT.parseArgsAndGo(ADT.java)
	at com.adobe.air.ADT.run(ADT.java)
	at com.adobe.air.ADT.main(ADT.java)
```

**Expected Result:**
Application will be packaged without errors. Or AIR SDK should contains requirements to Java version at least somewhere in Release Notes if some Java versions not supported.

## Known Workarounds
Use Java 8 or Java 11 (didn't test other).

**Comment by httpwebmedia** [comment]
I have the same issue on Java 11 and air .674 on windows 10 x64

**Comment by ajwfrost** [comment]
? It should be okay with Java 11, it's only the very recent JDKs/JREs where they're restricting access to the sun security classes..

Are you able to check, are you using ADT from the command line or via some other tool? as it might be that the other tool is somehow configured to use a later JDK by default and we'd then run into this problem... (I think we have seen this with Intellij ..)

Anyway -> we'll accelerate our work to get rid of this legacy internal-signing code and just rely on the Android tools. I think those classes aren't used in the AAB generation, in case you can switch targets to use aab rather than apk..

thanks

**Comment by httpwebmedia** [comment]
I try to create a new app on playstore and tried aab steps but it's impossible to get it works.
My environment : 
Windows 10 x64
Adobe Air .674
Java jdk-11.0.13

sdkmanager --list : 
```
  Path                                                | Version | Description                                     | Location
  -------                                             | ------- | -------                                         | -------
  build-tools;30.0.3                                  | 30.0.3  | Android SDK Build-Tools 30.0.3                  | build-tools\30.0.3
  emulator                                            | 30.9.5  | Android Emulator                                | emulator
  extras;android;m2repository                         | 47.0.0  | Android Support Repository                      | extras\android\m2repository
  extras;google;m2repository                          | 58      | Google Repository                               | extras\google\m2repository
  extras;intel;Hardware_Accelerated_Execution_Manager | 7.6.5   | Intel x86 Emulator Accelerator (HAXM installer) | extras\intel\Hardware_Accelerated_Execution_Manager
  patcher;v4                                          | 1       | SDK Patch Applier v4                            | patcher\v4
  platform-tools                                      | 31.0.3  | Android SDK Platform-Tools                      | platform-tools
  platforms;android-30                                | 3       | Android SDK Platform 30                         | platforms\android-30
  sources;android-29                                  | 1       | Sources for Android 29                          | sources\android-29
  tools                                               | 26.1.1  | Android SDK Tools                               | tools
```

Android SDK path : C:\Users\Default\AppData\Local\Android\Sdk\

on Command : echo %JAVA_HOME%
C:\Program Files\Java\jdk-11.0.13

So I try to compile on command  : 
```cd C:\_APPS\my-new-app && set AIR_NOANDROIDFLAIR=true && C:\Users\Default\Documents\AIRSDK_Windows_v33.1.1.674\bin\adt -package -target aab -storetype pkcs12 -keystore "C:\Dropbox\certificates\apps-certs-google-play-aab.p12" -storepass mypwd my-new-app-64.aab my-new-app-app.xml -extdir "C:\Dropbox\libs" -platformsdk C:\\Users\\Default\\AppData\\Local\\Android\\Sdk\\ my-new-app.swf icon36x36.png icon144x144.png icon96x96.png icon72x72.png icon48x48.png icon192x192.png```

and getting that output after a few minutes : 
```Warning: packagedDependency classes.jar used by more than one ANE. Ignoring this entry
Exception in thread "main" java.lang.IllegalAccessError: class com.adobe.air.apk.APKSigner (in unnamed module @0x261d8190) cannot access class sun.security.x509.X500Name (in module java.base) because module java.base does not export sun.security.x509 to unnamed module @0x261d8190
        at com.adobe.air.apk.APKSigner.getSignatureBlockFile(APKSigner.java)
        at com.adobe.air.apk.AABSignedFile.finalizeSig(AABSignedFile.java)
        at com.adobe.air.apk.AABPackager.renameOutput(AABPackager.java)
        at com.adobe.air.ApplicationPackager.createPackage(ApplicationPackager.java)
        at com.adobe.air.apk.AABPackager.createPackage(AABPackager.java)
        at com.adobe.air.ADT.parseArgsAndGo(ADT.java)
        at com.adobe.air.ADT.run(ADT.java)
        at com.adobe.air.ADT.main(ADT.java)```

I tried with java 17 but same issue... 

What am I missing?

Thank you for your help!

**Comment by ajwfrost** [comment]
Strange that Java 11 is throwing this error .. but for an AAB file, you don't actually need to sign it, so you can try:

```
cd C:\_APPS\my-new-app && set AIR_NOANDROIDFLAIR=true && C:\Users\Default\Documents\AIRSDK_Windows_v33.1.1.674\bin\adt -package -target aab my-new-app-64.aab my-new-app-app.xml -extdir "C:\Dropbox\libs" -platformsdk C:\\Users\\Default\\AppData\\Local\\Android\\Sdk\\ my-new-app.swf icon36x36.png icon144x144.png icon96x96.png icon72x72.png icon48x48.png icon192x192.png
```

thanks

**Comment by httpwebmedia** [comment]
Did not try yet but I need to sign for production...



Le 25 nov. 2021 à 16:51, à 16:51, Andrew Frost ***@***.***> a écrit:

**Comment by httpwebmedia** [comment]
Okay now it worked using your command but my issue is how do I sign my aab file then for production???

**Comment by idzdigital** [comment]
Same issue

**Comment by ajwfrost** [comment]
Bundle files are actually just signed using jarsigner:
https://docs.oracle.com/javase/7/docs/technotes/tools/windows/jarsigner.html

Also see the second half of https://airsdk.dev/docs/tutorials/platform/android/packaging-android-app-bundles for more details on signing, plus Google's notes at https://developer.android.com/studio/publish/app-signing

thanks

**Comment by ajwfrost** [comment]
FYI you can also try the updated version of ADT that we've just uploaded to:
https://transfer.harman.com/message/DG6Fk3Kq8je7h53CeglkmD

thanks

**Comment by httpwebmedia** [comment]
Ok finally it worked using the new adt.jar with air .686

It's a new app so I created  a new keystore (jks). 
I debug with a .p12 cert and it worked on the device with air .686 (debug through animate) but then I compiled and installed the apks on the same device and launched it it crashes right away with following device logs : 


`11-26 16:26:51.073  2987  2987 I zygote  : Rejecting re-init on previously-failed class java.lang.Class<androidx.work.impl.utils.futures.AbstractFuture>: java.lang.NoClassDefFoundError: Failed resolution of: Lcom/google/common/util/concurrent/ListenableFuture;
11-26 16:26:51.073  2987  2987 I zygote  :   at void androidx.work.impl.WorkManagerImpl.initialize(android.content.Context, androidx.work.Configuration) (WorkManagerImpl.java:181)
11-26 16:26:51.073  2987  2987 I zygote  :   at void androidx.work.WorkManager.initialize(android.content.Context, androidx.work.Configuration) (WorkManager.java:210)
11-26 16:26:51.073  2987  2987 I zygote  :   at boolean androidx.work.impl.WorkManagerInitializer.onCreate() (WorkManagerInitializer.java:40)
11-26 16:26:51.073  2987  2987 I zygote  :   at void android.content.ContentProvider.attachInfo(android.content.Context, android.content.pm.ProviderInfo, boolean) (ContentProvider.java:1925)
11-26 16:26:51.073  2987  2987 I zygote  :   at void android.content.ContentProvider.attachInfo(android.content.Context, android.content.pm.ProviderInfo) (ContentProvider.java:1900)
11-26 16:26:51.073  2987  2987 I zygote  :   at android.app.ContentProviderHolder android.app.ActivityThread.installProvider(android.content.Context, android.app.ContentProviderHolder, android.content.pm.ProviderInfo, boolean, boolean, boolean) (ActivityThread.java:6618)
11-26 16:26:51.073  2987  2987 I zygote  :   at void android.app.ActivityThread.installContentProviders(android.content.Context, java.util.List) (ActivityThread.java:6173)
11-26 16:26:51.073  2987  2987 I zygote  :   at void android.app.ActivityThread.handleBindApplication(android.app.ActivityThread$AppBindData) (ActivityThread.java:6079)
11-26 16:26:51.073  2987  2987 I zygote  :   at void android.app.ActivityThread.-wrap1(android.app.ActivityThread, android.app.ActivityThread$AppBindData) (ActivityThread.java:-1)
11-26 16:26:51.073  2987  2987 I zygote  :   at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1791)
11-26 16:26:51.073  2987  2987 I zygote  :   at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:106)
11-26 16:26:51.073  2987  2987 I zygote  :   at void android.os.Looper.loop() (Looper.java:164)
11-26 16:26:51.073  2987  2987 I zygote  :   at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:7000)
11-26 16:26:51.073  2987  2987 I zygote  :   at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
11-26 16:26:51.073  2987  2987 I zygote  :   at void com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run() (RuntimeInit.java:441)
11-26 16:26:51.073  2987  2987 I zygote  :   at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:1408)
11-26 16:26:51.073  2987  2987 I zygote  : Caused by: java.lang.ClassNotFoundException: Didn't find class "com.google.common.util.concurrent.ListenableFuture" on path: DexPathList[[zip file "/data/app/myapp.com-TQkivqyW40grhMgvPi0ruw==/base.apk", zip file "/data/app/myapp.com-TQkivqyW40grhMgvPi0ruw==/split_config.armeabi_v7a.apk", zip file "/data/app/myapp.com-TQkivqyW40grhMgvPi0ruw==/split_config.fr.apk", zip file "/data/app/myapp.com-TQkivqyW40grhMgvPi0ruw==/split_config.hdpi.apk"],nativeLibraryDirectories=[/data/app/myapp.com-TQkivqyW40grhMgvPi0ruw==/lib/arm, /data/app/myapp.com-TQkivqyW40grhMgvPi0ruw==/base.apk!/lib/armeabi-v7a, /data/app/myapp.com-TQkivqyW40grhMgvPi0ruw==/split_config.armeabi_v7a.apk!/lib/armeabi-v7a, /data/app/myapp.com-TQkivqyW40grhMgvPi0ruw==/split_config.fr.apk!/lib/armeabi-v7a, /data/app/
11-26 16:26:51.073  2987  2987 I zygote  :   at java.lang.Class dalvik.system.BaseDexClassLoader.findClass(java.lang.String) (BaseDexClassLoader.java:125)
11-26 16:26:51.073  2987  2987 I zygote  :   at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String, boolean) (ClassLoader.java:379)
11-26 16:26:51.073  2987  2987 I zygote  :   at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String) (ClassLoader.java:312)
11-26 16:26:51.073  2987  2987 I zygote  :   at void androidx.work.impl.WorkManagerImpl.initialize(android.content.Context, androidx.work.Configuration) (WorkManagerImpl.java:181)
11-26 16:26:51.073  2987  2987 I zygote  :   at void androidx.work.WorkManager.initialize(android.content.Context, androidx.work.Configuration) (WorkManager.java:210)
11-26 16:26:51.073  2987  2987 I zygote  :   at boolean androidx.work.impl.WorkManagerInitializer.onCreate() (WorkManagerInitializer.java:40)
11-26 16:26:51.073  2987  2987 I zygote  :   at void android.content.ContentProvider.attachInfo(android.content.Context, android.content.pm.ProviderInfo, boolean) (ContentProvider.java:1925)
11-26 16:26:51.073  2987  2987 I zygote  :   at void android.content.ContentProvider.attachInfo(android.content.Context, android.content.pm.ProviderInfo) (ContentProvider.java:1900)
11-26 16:26:51.073  2987  2987 I zygote  :   at android.app.ContentProviderHolder android.app.ActivityThread.installProvider(android.content.Context, android.app.ContentProviderHolder, android.content.pm.ProviderInfo, boolean, boolean, boolean) (ActivityThread.java:6618)
11-26 16:26:51.074  2987  2987 I zygote  :   at void android.app.ActivityThread.installContentProviders(android.content.Context, java.util.List) (ActivityThread.java:6173)
11-26 16:26:51.074  2987  2987 I zygote  :   at void android.app.ActivityThread.handleBindApplication(android.app.ActivityThread$AppBindData) (ActivityThread.java:6079)
11-26 16:26:51.074  2987  2987 I zygote  :   at void android.app.ActivityThread.-wrap1(android.app.ActivityThread, android.app.ActivityThread$AppBindData) (ActivityThread.java:-1)
11-26 16:26:51.074  2987  2987 I zygote  :   at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1791)
11-26 16:26:51.074  2987  2987 I zygote  :   at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:106)
11-26 16:26:51.074  2987  2987 I zygote  :   at void android.os.Looper.loop() (Looper.java:164)
11-26 16:26:51.074  2987  2987 I zygote  :   at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:7000)
11-26 16:26:51.074  2987  2987 I zygote  :   at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
11-26 16:26:51.074  2987  2987 I zygote  :   at void com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run() (RuntimeInit.java:441)
11-26 16:26:51.074  2987  2987 I zygote  :   at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:1408)
11-26 16:26:51.074  2987  2987 I zygote  :
11-26 16:26:51.074  2987  2987 I zygote  : Rejecting re-init on previously-failed class java.lang.Class<androidx.work.impl.utils.futures.SettableFuture>: java.lang.NoClassDefFoundError: Failed resolution of: Landroidx/work/impl/utils/futures/AbstractFuture;
11-26 16:26:51.075  2987  2987 I zygote  :   at void androidx.work.impl.WorkManagerImpl.initialize(android.content.Context, androidx.work.Configuration) (WorkManagerImpl.java:181)
11-26 16:26:51.075  2987  2987 I zygote  :   at void androidx.work.WorkManager.initialize(android.content.Context, androidx.work.Configuration) (WorkManager.java:210)
11-26 16:26:51.075  2987  2987 I zygote  :   at boolean androidx.work.impl.WorkManagerInitializer.onCreate() (WorkManagerInitializer.java:40)
11-26 16:26:51.075  2987  2987 I zygote  :   at void android.content.ContentProvider.attachInfo(android.content.Context, android.content.pm.ProviderInfo, boolean) (ContentProvider.java:1925)
11-26 16:26:51.075  2987  2987 I zygote  :   at void android.content.ContentProvider.attachInfo(android.content.Context, android.content.pm.ProviderInfo) (ContentProvider.java:1900)
11-26 16:26:51.075  2987  2987 I zygote  :   at android.app.ContentProviderHolder android.app.ActivityThread.installProvider(android.content.Context, android.app.ContentProviderHolder, android.content.pm.ProviderInfo, boolean, boolean, boolean) (ActivityThread.java:6618)
11-26 16:26:51.075  2987  2987 I zygote  :   at void android.app.ActivityThread.installContentProviders(android.content.Context, java.util.List) (ActivityThread.java:6173)
11-26 16:26:51.075  2987  2987 I zygote  :   at void android.app.ActivityThread.handleBindApplication(android.app.ActivityThread$AppBindData) (ActivityThread.java:6079)
11-26 16:26:51.075  2987  2987 I zygote  :   at void android.app.ActivityThread.-wrap1(android.app.ActivityThread, android.app.ActivityThread$AppBindData) (ActivityThread.java:-1)
11-26 16:26:51.075  2987  2987 I zygote  :   at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1791)
11-26 16:26:51.075  2987  2987 I zygote  :   at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:106)
11-26 16:26:51.075  2987  2987 I zygote  :   at void android.os.Looper.loop() (Looper.java:164)
11-26 16:26:51.075  2987  2987 I zygote  :   at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:7000)
11-26 16:26:51.075  2987  2987 I zygote  :   at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
11-26 16:26:51.075  2987  2987 I zygote  :   at void com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run() (RuntimeInit.java:441)
11-26 16:26:51.075  2987  2987 I zygote  :   at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:1408)
11-26 16:26:51.075  2987  2987 I zygote  : Caused by: java.lang.ClassNotFoundException: androidx.work.impl.utils.futures.AbstractFuture
11-26 16:26:51.075  2987  2987 I zygote  :   at java.lang.Class java.lang.VMClassLoader.findLoadedClass(java.lang.ClassLoader, java.lang.String) (VMClassLoader.java:-2)
11-26 16:26:51.075  2987  2987 I zygote  :   at java.lang.Class java.lang.ClassLoader.findLoadedClass(java.lang.String) (ClassLoader.java:738)
11-26 16:26:51.075  2987  2987 I zygote  :   at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String, boolean) (ClassLoader.java:363)
11-26 16:26:51.075  2987  2987 I zygote  :   at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String) (ClassLoader.java:312)
11-26 16:26:51.075  2987  2987 I zygote  :   at void androidx.work.impl.WorkManagerImpl.initialize(android.content.Context, androidx.work.Configuration) (WorkManagerImpl.java:181)
11-26 16:26:51.075  2987  2987 I zygote  :   at void androidx.work.WorkManager.initialize(android.content.Context, androidx.work.Configuration) (WorkManager.java:210)
11-26 16:26:51.075  2987  2987 I zygote  :   at boolean androidx.work.impl.WorkManagerInitializer.onCreate() (WorkManagerInitializer.java:40)
11-26 16:26:51.075  2987  2987 I zygote  :   at void android.content.ContentProvider.attachInfo(android.content.Context, android.content.pm.ProviderInfo, boolean) (ContentProvider.java:1925)
11-26 16:26:51.075  2987  2987 I zygote  :   at void android.content.ContentProvider.attachInfo(android.content.Context, android.content.pm.ProviderInfo) (ContentProvider.java:1900)
11-26 16:26:51.075  2987  2987 I zygote  :   at android.app.ContentProviderHolder android.app.ActivityThread.installProvider(android.content.Context, android.app.ContentProviderHolder, android.content.pm.ProviderInfo, boolean, boolean, boolean) (ActivityThread.java:6618)
11-26 16:26:51.075  2987  2987 I zygote  :   at void android.app.ActivityThread.installContentProviders(android.content.Context, java.util.List) (ActivityThread.java:6173)
11-26 16:26:51.075  2987  2987 I zygote  :   at void android.app.ActivityThread.handleBindApplication(android.app.ActivityThread$AppBindData) (ActivityThread.java:6079)
11-26 16:26:51.075  2987  2987 I zygote  :   at void android.app.ActivityThread.-wrap1(android.app.ActivityThread, android.app.ActivityThread$AppBindData) (ActivityThread.java:-1)
11-26 16:26:51.075  2987  2987 I zygote  :   at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1791)
11-26 16:26:51.075  2987  2987 I zygote  :   at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:106)
11-26 16:26:51.075  2987  2987 I zygote  :   at void android.os.Looper.loop() (Looper.java:164)
11-26 16:26:51.075  2987  2987 I zygote  :   at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:7000)
11-26 16:26:51.075  2987  2987 I zygote  :   at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
11-26 16:26:51.075  2987  2987 I zygote  :   at void com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run() (RuntimeInit.java:441)
11-26 16:26:51.075  2987  2987 I zygote  :   at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:1408)
11-26 16:26:51.075  2987  2987 I zygote  :
11-26 16:26:51.076  2987  2987 I zygote  : Rejecting re-init on previously-failed class java.lang.Class<androidx.work.impl.utils.futures.SettableFuture>: java.lang.NoClassDefFoundError: Failed resolution of: Landroidx/work/impl/utils/futures/AbstractFuture;
11-26 16:26:51.076  2987  2987 I zygote  :   at void androidx.work.impl.WorkManagerImpl.initialize(android.content.Context, androidx.work.Configuration) (WorkManagerImpl.java:181)
11-26 16:26:51.076  2987  2987 I zygote  :   at void androidx.work.WorkManager.initialize(android.content.Context, androidx.work.Configuration) (WorkManager.java:210)
11-26 16:26:51.076  2987  2987 I zygote  :   at boolean androidx.work.impl.WorkManagerInitializer.onCreate() (WorkManagerInitializer.java:40)
11-26 16:26:51.076  2987  2987 I zygote  :   at void android.content.ContentProvider.attachInfo(android.content.Context, android.content.pm.ProviderInfo, boolean) (ContentProvider.java:1925)
11-26 16:26:51.076  2987  2987 I zygote  :   at void android.content.ContentProvider.attachInfo(android.content.Context, android.content.pm.ProviderInfo) (ContentProvider.java:1900)
11-26 16:26:51.076  2987  2987 I zygote  :   at android.app.ContentProviderHolder android.app.ActivityThread.installProvider(android.content.Context, android.app.ContentProviderHolder, android.content.pm.ProviderInfo, boolean, boolean, boolean) (ActivityThread.java:6618)
11-26 16:26:51.076  2987  2987 I zygote  :   at void android.app.ActivityThread.installContentProviders(android.content.Context, java.util.List) (ActivityThread.java:6173)
11-26 16:26:51.076  2987  2987 I zygote  :   at void android.app.ActivityThread.handleBindApplication(android.app.ActivityThread$AppBindData) (ActivityThread.java:6079)
11-26 16:26:51.076  2987  2987 I zygote  :   at void android.app.ActivityThread.-wrap1(android.app.ActivityThread, android.app.ActivityThread$AppBindData) (ActivityThread.java:-1)
11-26 16:26:51.076  2987  2987 I zygote  :   at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1791)
11-26 16:26:51.076  2987  2987 I zygote  :   at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:106)
11-26 16:26:51.076  2987  2987 I zygote  :   at void android.os.Looper.loop() (Looper.java:164)
11-26 16:26:51.076  2987  2987 I zygote  :   at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:7000)
11-26 16:26:51.076  2987  2987 I zygote  :   at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
11-26 16:26:51.076  2987  2987 I zygote  :   at void com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run() (RuntimeInit.java:441)
11-26 16:26:51.076  2987  2987 I zygote  :   at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:1408)
11-26 16:26:51.076  2987  2987 I zygote  : Caused by: java.lang.ClassNotFoundException: androidx.work.impl.utils.futures.AbstractFuture
11-26 16:26:51.076  2987  2987 I zygote  :   at java.lang.Class java.lang.VMClassLoader.findLoadedClass(java.lang.ClassLoader, java.lang.String) (VMClassLoader.java:-2)
11-26 16:26:51.076  2987  2987 I zygote  :   at java.lang.Class java.lang.ClassLoader.findLoadedClass(java.lang.String) (ClassLoader.java:738)
11-26 16:26:51.076  2987  2987 I zygote  :   at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String, boolean) (ClassLoader.java:363)
11-26 16:26:51.076  2987  2987 I zygote  :   at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String) (ClassLoader.java:312)
11-26 16:26:51.076  2987  2987 I zygote  :   at void androidx.work.impl.WorkManagerImpl.initialize(android.content.Context, androidx.work.Configuration) (WorkManagerImpl.java:181)
11-26 16:26:51.076  2987  2987 I zygote  :   at void androidx.work.WorkManager.initialize(android.content.Context, androidx.work.Configuration) (WorkManager.java:210)
11-26 16:26:51.076  2987  2987 I zygote  :   at boolean androidx.work.impl.WorkManagerInitializer.onCreate() (WorkManagerInitializer.java:40)
11-26 16:26:51.076  2987  2987 I zygote  :   at void android.content.ContentProvider.attachInfo(android.content.Context, android.content.pm.ProviderInfo, boolean) (ContentProvider.java:1925)
11-26 16:26:51.076  2987  2987 I zygote  :   at void android.content.ContentProvider.attachInfo(android.content.Context, android.content.pm.ProviderInfo) (ContentProvider.java:1900)
11-26 16:26:51.076  2987  2987 I zygote  :   at android.app.ContentProviderHolder android.app.ActivityThread.installProvider(android.content.Context, android.app.ContentProviderHolder, android.content.pm.ProviderInfo, boolean, boolean, boolean) (ActivityThread.java:6618)
11-26 16:26:51.076  2987  2987 I zygote  :   at void android.app.ActivityThread.installContentProviders(android.content.Context, java.util.List) (ActivityThread.java:6173)
11-26 16:26:51.076  2987  2987 I zygote  :   at void android.app.ActivityThread.handleBindApplication(android.app.ActivityThread$AppBindData) (ActivityThread.java:6079)
11-26 16:26:51.076  2987  2987 I zygote  :   at void android.app.ActivityThread.-wrap1(android.app.ActivityThread, android.app.ActivityThread$AppBindData) (ActivityThread.java:-1)
11-26 16:26:51.076  2987  2987 I zygote  :   at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1791)
11-26 16:26:51.076  2987  2987 I zygote  :   at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:106)
11-26 16:26:51.076  2987  2987 I zygote  :   at void android.os.Looper.loop() (Looper.java:164)
11-26 16:26:51.076  2987  2987 I zygote  :   at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:7000)
11-26 16:26:51.076  2987  2987 I zygote  :   at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
11-26 16:26:51.076  2987  2987 I zygote  :   at void com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run() (RuntimeInit.java:441)
11-26 16:26:51.076  2987  2987 I zygote  :   at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:1408)
11-26 16:26:51.076  2987  2987 I zygote  :
11-26 16:26:51.082  2987  3002 W ResourceType: ResTable_typeSpec entry count inconsistent: given 74, previously 76
11-26 16:26:51.099  2915  3120 D PackageManager: setEnabledSetting : userId = 0 packageName = myapp.com cmp = androidx.work.impl.background.systemjob.SystemJobService newState = 1 callingPackage = 13271/myapp.com
`

Your support (@ajwfrost ) is appreciated :)

**Comment by ajwfrost** [comment]
Hi @httpwebmedia 
Are you using a bundle there? as the failure seems to suggest there are split APKs that are installed from the creation of an App bundle?
```
Caused by: java.lang.ClassNotFoundException: Didn't find class "com.google.common.util.concurrent.ListenableFuture" on path:
DexPathList[[zip file "/data/app/myapp.com-TQkivqyW40grhMgvPi0ruw==/base.apk",
             zip file "/data/app/myapp.com-TQkivqyW40grhMgvPi0ruw==/split_config.armeabi_v7a.apk",
             zip file "/data/app/myapp.com-TQkivqyW40grhMgvPi0ruw==/split_config.fr.apk", 
etc
```

The underlying issue seems to be caused by a missing class definition .. maybe ANE related? or your app manifest isn't quite right perhaps?

Have you tried uninstalling the previous app from the handset, and maybe even rebooting it, before re-installing?

thanks

**Comment by httpwebmedia** [comment]
@ajwfrost Well I used bundletool to create and install the apks but nothing specific here. When I use debug to usb on animate the app works great. When I compile to aab file and use bundletool the app crashes right away when I attempt to launch it on the device... I've been trying for several days... it's first time in 8 years of doing air apps I struggle that much.

**Comment by ajwfrost** [comment]
Hi @httpwebmedia  - yes this App Bundle format has caused us a lot of problems, plus moving over to a Gradle-based build system has then added to the issues... Wondering if you'd be okay to send us the AAB file - preferably if you build with "aab-debug" target it woudl be good? And then, if you also have an APK that you generated via Animate, that would be useful too...
Upload link: https://transfer.harman.com/requests/jXI3DraXgJioiU7qIfdPnh

There must be a difference in the classes that are being supported and loaded in, but it's all Java-based errors which are in classes not used directly by AIR, hence my thinking that it's ANE-related. But we should be able to tell what's going on if we have one of the packages.

thanks

**Comment by httpwebmedia** [comment]
Thank you for your patience @ajwfrost

I managed to make it works (finally). It looks like you were right about ANE files. I updated some of the distriqt ANEs (admob, google play services and some more). 

To be continued :)

**Comment by ajwfrost** [comment]
> Thank you for your patience

Ditto!
thanks

**Comment by itlancer** [comment]
@ajwfrost 
With adt.jar patch https://github.com/airsdk/Adobe-Runtime-Support/issues/1368#issuecomment-979997595 Java 17 still doesn't work. The same error as described in original post.

**Comment by itlancer** [comment]
@ajwfrost 
With latest 33.1.1.698 APK packaging using Java 17 works only if in application descriptor added
`<BuildLegacyAPK>true</BuildLegacyAPK>`
Without `BuildLegacyAPK` setting it fails with the same error as AAB packaging: https://github.com/airsdk/Adobe-Runtime-Support/issues/1405

**Comment by ajwfrost** [comment]
@itlancer  thanks -> so yes, if you have the legacy build settings then it would work because it doesn't use gradle (and because we took out those sun-internal classes that were causing the earlier problem).

For the new gradle-based build, we need to update to Gradle 7.3 per https://github.com/airsdk/Adobe-Runtime-Support/issues/1405#issuecomment-984795000

thanks

**Comment by itlancer** [comment]
@ajwfrost 
Fixing `gradle-wrapper.properties` did the trick https://github.com/airsdk/Adobe-Runtime-Support/issues/1405#issuecomment-984795000. Now it works as expected.
Thanks! Waiting for a new AIR SDK release to close this issue.

**Comment by itlancer** [comment]
Fixed with latest AIR 33.1.1.713.
Thanks!
